src/SingleBookingPage/bookingItem/CarrierLogoWrapper.js

```diff
@@ -12,11 +12,19 @@ type Props = {
 };
 
 const CarrierLogoWrapper = (props: Props) => {
+  const addedCarriers = {};
   const carriers = (props.legs || [])
-    .map(leg => {
+    .filter(leg => {
```

Comment:
Not a crazy critical edge case that needs great optimization, but what about something like this? I personally find it more clear:
```javascript
const CarrierLogoWrapper = (props: Props) => {
  const carriers = {};
  (props.legs || []).forEach(leg => {
    const code = idx(leg, _ => _.airline.code);
    const name = idx(leg, _ => _.airline.name);
    if (code && name && !carriers[code]) {
      carriers[code] = { code, name };
    }
  });

  return <CarrierLogo size="large" carriers={Object.values(carriers)} />;
};
```

We could make it all within one loop, but `Object.values` is quite explicit :)

Reply:
Looks better then my filter/map/filter maze!

Reply:
The only problem with this approach is flow.. https://github.com/facebook/flow/issues/2221
So I made it like this `const carriersArray = Object.keys(carriers).map(key => carriers[key]);`
Do you maybe have any suggestions?

Reply:
Oh the flow, that great frie-nemy. 
I didn't try, but a mix of your approach and mine maybe? Avoiding the `Object.values` loop and hence super-mega-optimizing this algorithm to a single loop :roll_eyes: 
```javascript
const CarrierLogoWrapper = (props: Props) => {
  const carriers = {};
  const finalCarriers = [];
  (props.legs || []).forEach(leg => {
    const code = idx(leg, _ => _.airline.code);
    const name = idx(leg, _ => _.airline.name);
    if (code && name && !carriers[code]) {
      carriers[code] = true;
      finalCarriers.push({ code, name });
    }
  });

  return <CarrierLogo size="large" carriers={finalCarriers} />;
};
```

Reply:
We have a battle tested algorithm now 😎